project-service/api.md

```diff
@@ -0,0 +1,594 @@
+  + `GET`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + None
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+[
+  {
+    id: 1,
+    name: 'mvp'
+    status: 'Started',
+    startDate: [date],
+    endDate: [date]
+  }, {
+    id: 2,
+    name: 'profit',
+    status: 'Not Started',
+    startDate: [date],
+    endDate: [date]
+  }
+]
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Create Sprint
+
+Create a new sprint for a project
+
+- URL
+  + `/sprints`
+- Method
+  + `POST`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + Required
+    * `name=[string]`
+    * `status=[string]`
+    * `startDate=[date]`
+    * `endDate=[date]`
+- Success Response
+  + Code: `201 CREATED`
+  + Content:
+
+```
+{
+  id: 1,
+  name: 'extra features',
+  status: 'Not Started',
+  startDate: [date],
+  endDate: [date],
+  createdAt: [date]
+}
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Edit Sprint
+
+Edit details of a sprint
+
+- URL
+  + `/sprints`
+- Method
+  + `PUT`
+- URL Params
+  + Required
+    * `sprintId=[number]`
+- Data Params
+  + Optional
+    * `name=[string]`
+    * `status=[string]`
+    * `startDate=[date]`
+    * `endDate=[date]`
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+{
+  id: 1,
+  name: 'extra features',
+  status: 'Started',
+  startDate: [date],
+  endDate: [date],
+  createdAt: [date]
+}
+```
+
+- Error Response
+  + Invalid Sprint
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Sprint doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Delete Sprint
+
+Delete a sprint (tasks that belong to this sprint are not deleted)
+
+- URL
+  + `/sprints`
+- Method
+  + `DELETE`
+- URL Params
+  + Required
+    * `sprintId=[number]`
+- Data Params
+  + None
+- Success Response
+  + Code: `204 NO CONTENT`
+  + Content: None
+- Error Response
+  + Invalid Sprint
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Sprint doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+## Tasks
+
+### Fetch Tasks for a Project
+
+Get all tasks for a project
+
+- URL
+  + `/tasks`
+- Method
+  + `GET`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + None
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+[
+  {
+    id: 1,
+    name: 'do a task',
+    description: 'Ready',
+    userId: 2,
+    sprintId: 4
```

Comment:
Ok, a task needs to have a status
In the backlog, we will display only the ones that are yet to be done and that are not in the current sprint, so no need to send all tasks perhaps.
But maybe there will be a view for all tasks, or for completed tasks as well.